src/libraries/System.Text.Json/codegenerator/codegenerator.csproj

```diff
@@ -0,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
This is a shipping assembly and will need a better name.

Reply:
How's System.Text.Json.CodeGeneration? We should update the test projects as well.

Reply:
I have conflict with the name CodeGeneration tbh.
The way I see it SourceGenerator would call CodeGeneration which would be the literal string generator from System.Type (tbd).

Reply:
> The way I see it SourceGenerator would call CodeGeneration which would be the literal string generator from System.Type (tbd).

Internal type factoring aside, I think we should only have 1 assembly for this component.

@kevinwkt can you suggest a better name?

Reply:
So I was thinking 
SourceGeneration
  -> SourceGenerator
  -> CodeGenerator

what do you think @layomia @ericstj ?

Reply:
I think it should describe what is doing i.e  SourceGeneration, similar to how we have `System.Text.Json.Serialization`.

Reply:
It needs to have the System.Text.Json prefix, I don't have much of an opinion after that.  Here's what the guidelines have to say: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-assemblies-and-dlls  Doesn't exactly clear anything up :)  It's not too important since this should really only ever be seen in debugging.  Users should never see this name.